shipyard-controller/db/migration/project_credentials_migration.go

```diff
@@ -0,0 +1,59 @@
+package migration
+
+import (
+	"fmt"
+
+	"github.com/keptn/keptn/shipyard-controller/common"
+	"github.com/keptn/keptn/shipyard-controller/db"
+	"github.com/keptn/keptn/shipyard-controller/models"
+)
+
+func NewProjectCredentialsMigrator(dbConnection *db.MongoDBConnection, secretStore common.SecretStore) *ProjectCredentialsMigrator {
+	return &ProjectCredentialsMigrator{
+		projectRepo: db.NewMongoDBProjectCredentialsRepo(dbConnection),
+		secretRepo:  db.NewMongoDBSecretCredentialsRepo(secretStore),
+	}
+}
+
+type ProjectCredentialsMigrator struct {
+	projectRepo db.MongoDBProjectCredentialsRepo
+	secretRepo  db.MongoDBSecretCredentialsRepo
+}
+
+func (s *ProjectCredentialsMigrator) Transform() error {
+	projects, err := s.projectRepo.GetOldCredentialsProjects()
+	if err != nil {
+		return fmt.Errorf("could not transform git credentials to new format: %s", err.Error())
+	}
+	if err := s.UpdateSecrets(projects); err != nil {
+		return err
+	}
+	return s.UpdateProjects(projects)
+
+}
+
+func (s *ProjectCredentialsMigrator) UpdateProjects(projects []*models.ExpandedProjectOld) error {
+	if projects == nil {
+		return nil
+	}
+	for _, project := range projects {
+		err := s.projectRepo.UpdateProject(project)
```

Comment:
will this always do an update of every project whenever the pod is restarted, or do we have a check that prevents the project from being updated if it is already using the current model?

Reply:
we do not have a check, it's updated every time the pod is restarted, but it's made the same way for all migrations AFAIK